### PR TITLE
Truncate stack overflow error to avoid app from crashing forever

### PR DESCRIPTION
**Pull request by guperrot**
Keep 128 first frames and 128 last frames for huge exceptions.
Add that test in sasquatch and integration test.

### CODE REVIEW

File: sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/utils/ErrorLogHelper.java [hunk 1]
```diff
@@ -229,14 +241,29 @@ private static Exception getModelExceptionFromThrowable(@NonNull Throwable t) {
...
-            stackFrame.setFileName(stackTraceElement.getFileName());
-            stackFrames.add(stackFrame);
+        if (stackTrace.length > FRAME_LIMIT) {
+            for (int i = 0; i < FRAME_LIMIT_HALF; i++) {
+                stackFrames.add(getModelStackFrame(stackTrace[i]));
+            }
+            for (int i = stackTrace.length - FRAME_LIMIT_HALF; i < stackTrace.length; i++) {
+                stackFrames.add(getModelStackFrame(stackTrace[i]));
+            }
+            Log.w(Crashes.LOG_TAG, "Crash frames truncated from " + stackTrace.length + " to " + stackFrames.size() + " frames.");
```
Comment (jaelim-ms) [inline sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/utils/ErrorLogHelper.java]: MobileCenterLog?

File: sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/utils/ErrorLogHelper.java [hunk 2]
```diff
@@ -44,6 +45,17 @@
     static final String ERROR_DIRECTORY = "error";
 
     /**
+     * For huge stack traces such as giant StackOverflowError, we keep only beginning and end of frames according to this limit.
+     */
+    @VisibleForTesting
```
Comment (jaelim-ms) [inline sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/utils/ErrorLogHelper.java]: It doesn't do anything exception documentation. You can move this to Crashes, remove the annotation or leave as it is.

File: sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/utils/ErrorLogHelper.java (hunk 1 above)
Comment (guperrot) [inline sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/utils/ErrorLogHelper.java]: oops

File: sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/utils/ErrorLogHelper.java (hunk 2 above)
Comment (guperrot) [inline sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/utils/ErrorLogHelper.java]: Need it to be public for test, hence annotation.

**Pull request merged**